README.rst

```diff
@@ -0,0 +1,45 @@
+Blockstore Client
+=================
+
+This package contains a python REST API client library for the Open edX Blockstore REST API. It also contains a command line tool that makes it easy to work with Blockstore from the command line.
+
+This package supports Python 2 and 3.
+
+Details
+-------
+
+The API client code in this repository is completely auto-generated from Blockstore's OpenAPI specification file, using `openapi-generator <https://github.com/OpenAPITools/openapi-generator>`_. Do not edit the code in `blockstore_api`, as your changes will be lost. Instead, update the Blockstore API spec file, then run `make api-client` in this repository to update the auto-generated code.
+
+Usage Example
+-------------
+
+Using the Tagstore API::
+
+    >>> from blockstore_api import TagstoreApi, Taxonomy, TagWithHierarchy, Entity
+    >>> tagstore = TagstoreApi()
+    >>> tax = tagstore.taxonomies_create(Taxonomy(name="Example Taxonomy"))
+    >>> tax.id
+    42
+    >>> veg_tag = tagstore.taxonomies_tags_create(tax.id, TagWithHierarchy(name="Vegetable"))
+    >>> carrot_tag = tagstore.taxonomies_tags_create(tax.id, TagWithHierarchy(name="Carrot", parent=veg_tag.name))
```

Comment:
This client library feels overly verbose and too exposing of it internals. Are we limited to this because of technical reasons?

For example, if I were writing client code to call a tagging service, I'd want to write something like:

```
from blockstore_api import TagstoreApi
tagstore = TagstoreApi()
taxonomy = tagstore.create("Example Taxonomy")
tag1 = tagstore.create(taxonomy, "Vegetable")
tag2 = tagstore.create(taxonomy, "Carrot", parent=tag1)
```

If I have to think too hard about the structure of the api, like `TagWithHierarchy(name="Carrot", parent=veg_tag.name)`, (v.s. the structure of the structure of _my_ data in the context I am using it), then I start to feel like I could just as easily use requests and talk to the api myself. What do you think?

Reply:
Part of the reason I'm using an auto-generated client here is to ensure that the API spec (which is a key part of the API documentation) is complete. If the auto-generated API client works, then the spec is accurate, and if it doesn't, then the spec is inaccurate. Convenience is a secondary consideration, and certainly if this approach proves to be significantly less convenient than using requests directly, people should just use requests directly.

That said, the latest version of this that I've developed is closer to what you wrote:

```python
from blockstore_api import TagstoreApi, Taxonomy, Tag
tagstore = TagstoreApi()
tax = tagstore.create_taxonomy(Taxonomy(name="Computer Science Taxonomy"))
language_tag = tagstore.add_taxonomy_tag(tax.id, Tag(name="Programming Languages"))
python_tag = tagstore.add_taxonomy_tag(tax.id, Tag(name="Python", parent=language_tag.name))
```